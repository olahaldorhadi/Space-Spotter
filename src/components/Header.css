.header {
    background-color: var(--dark-blue);
    width: 100%;
    height: 150px;
    top: 0;
    display: flex;
    align-items: center;
}

.header-content {
    height: 90%;
    margin-left: auto;
    margin-right: auto;
    display: flex;
    flex-direction: row;
    align-items: center;
}

.logo {
    height: 100px; 
    width: auto; 
  }

.header-content img {
    height: 100%;
    width:fit-content;
}

.explanation-text {
    margin-left: 20px;
    text-align: left;
}

.explanation-text h1, .explanation-text p {
    margin-top: 10px;
    margin-bottom: 10px;
}