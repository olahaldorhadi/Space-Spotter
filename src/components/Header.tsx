import './Header.css'
import Logo from '../assets/SpaceSpotterLogo.png'

const Header = () => {
    return (
        <>
            <div className="header">
                <div className="header-content">
                    <img src={Logo} />
                    <div className="explanation-text">
                        <h1>Trenger du et rom?</h1>
                        <p>Finn oversikt over rom p√• din campus!</p>
                    </div>
                </div>
            </div>
        </>
    )
}

export default Header
