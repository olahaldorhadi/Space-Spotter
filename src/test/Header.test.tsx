import { describe, expect, it } from 'vitest'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Header from '../components/Header'

describe('Header component', () => {
    it('renders Header component', () => {
        render(<Header />)

        // Check if the logo is present on the page
        const logoElement = screen.getByRole('img')
        expect(logoElement).toBeInTheDocument()

        // Check if the header text is present
        const headerTextElement = screen.getByRole('heading', { level: 1 })
        expect(headerTextElement).toHaveTextContent('Trenger du et rom?')

        // Check if the explanation text is present
        const explanationTextElement = screen.getByText(
            'Finn oversikt over rom p√• din campus!'
        )
        expect(explanationTextElement).toBeInTheDocument()

        // Create a "snapshot" by comparing the rendered HTML
        const headerComponentHTML = document.body.innerHTML
        expect(headerComponentHTML).toMatchSnapshot()
    })
})
