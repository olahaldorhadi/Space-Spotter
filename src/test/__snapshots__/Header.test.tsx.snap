// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders Header component 1`] = `"<div><div class=\\"header\\"><div class=\\"header-content\\"><img src=\\"/project1/src/assets/SpaceSpotterLogo.png\\"><div class=\\"explanation-text\\"><h1>Trenger du et rom?</h1><p>Finn oversikt over rom pÃ¥ din campus!</p></div></div></div></div>"`;
